File: src/main/java/ar/edu/um/tesoreria/rest/model/Ejercicio.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;
import java.time.OffsetDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.fasterxml.jackson.annotation.JsonFormat;

import ar.edu.um.tesoreria.rest.util.Tool;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "ejercicios")
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Ejercicio extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 2680545663863027580L;

	@Id
	@Column(name = "eje_id")
	private Integer ejercicioId;

	@Column(name = "eje_nombre")
	private String nombre;

	@Column(name = "eje_fechainicio")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fechaInicio = Tool.dateAbsoluteArgentina();

	@Column(name = "eje_fechafin")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fechaFinal = Tool.dateAbsoluteArgentina();

	@Column(name = "eje_bloqueado")
	private Byte bloqueado;

	private Integer ordenContableResultado;
	private Integer ordenContableBienesUso;

}

File: src/main/java/um/tesoreria/rest/kotlin/model/Ejercicio.kt
package um.tesoreria.rest.kotlin.model

import um.tesoreria.rest.util.Tool
import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.OffsetDateTime

@Entity
@Table(name = "ejercicios")
data class Ejercicio(

    @Id
    @Column(name = "eje_id")
    var ejercicioId: Int? = null,

    @Column(name = "eje_nombre")
    var nombre: String? = null,

    @Column(name = "eje_fechainicio")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaInicio: OffsetDateTime? = Tool.dateAbsoluteArgentina(),

    @Column(name = "eje_fechafin")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaFinal: OffsetDateTime? = Tool.dateAbsoluteArgentina(),

    @Column(name = "eje_bloqueado")
    var bloqueado: Byte = 0,

    var ordenContableResultado: Int? = null,
    var ordenContableBienesUso: Int? = null

) : Auditable()
